git/config.py

```diff
@@ -443,7 +452,49 @@ def string_decode(v):
             raise e
 
     def _has_includes(self):
-        return self._merge_includes and self.has_section('include')
+        return self._merge_includes and len(self._included_paths())
+
+    def _included_paths(self):
+        """Return all paths that must be included to configuration.
+        """
+        paths = []
+
+        for section in self.sections():
+            if section == "include":
+                paths += self.items(section)
+
+            match = CONDITIONAL_INCLUDE_REGEXP.search(section)
+            if match is not None and self._repo is not None:
```

Comment:
I think it would be better to 'continue' if one of the prerequisites isn't met, reducing the indentation level by one.
Along the lines of this:
```python
if match is None or self._repo is None:
    continue
```

Reply:
True, I'm gonna change that

Reply:
Changed